samples/portable-sdks/python/sub-orchestrations-with-fan-out-fan-in/worker.py

```diff
@@ -0,0 +1,147 @@
+
+    # Ship order
+    order_shipped = yield ctx.call_activity('ship_order', input=order)
+
+    if not order_shipped:
+        return {'order': order, 'status': 'failed', 'reason': 'shipping failed'}
+
+    # Notify customer
+    customer_notified = yield ctx.call_activity('notify_customer', input=order)
+
+    if not customer_notified:
+        return {'order': order, 'status': 'failed', 'reason': 'customer notification failed'}
+
+    # Return success status
+    return {'order': order, 'status': 'completed'}
+
+def orchestrator(ctx, _):
+    """Orchestrator function that calls the 'get_orders' and 'process_order'
+    sub-orchestration functions in parallel, waits for them all to complete, and prints
+    an aggregate summary of the outputs"""
+
+    orders: list[str] = yield ctx.call_activity('get_orders')
+
+    # Execute the orders in parallel and wait for them all to return
+    tasks = [ctx.call_sub_orchestrator(process_order, input=order) for order in orders]
+    results: list[dict] = yield task.when_all(tasks)
+
+    # Return an aggregate summary of the results
+    return {
+        'orders': orders,
+        'results': results,
+        'total_completed': sum(1 for result in results if result['status'] == 'completed'),
+        'total_failed': sum(1 for result in results if result['status'] == 'failed'),
+        'details': results,
+    }
+
+# Read the environment variable
+taskhub_name = os.getenv("TASKHUB")
+
+# Check if the variable exists
+if taskhub_name:
+    print(f"The value of TASKHUB is: {taskhub_name}")
+else:
+    print("TASKHUB is not set. Please set the TASKHUB environment variable to the name of the taskhub you wish to use")
+    print("If you are using windows powershell, run the following: $env:TASKHUB=\"<taskhubname>\"")
+    print("If you are using bash, run the following: export TASKHUB=\"<taskhubname>\"")
+    exit()
+
+# Read the environment variable
+endpoint = os.getenv("ENDPOINT")
+
+# Check if the variable exists
+if endpoint:
+    print(f"The value of ENDPOINT is: {endpoint}")
+else:
+    print("ENDPOINT is not set. Please set the ENDPOINT environment variable to the endpoint of the scheduler")
+    print("If you are using windows powershell, run the following: $env:ENDPOINT=\"<schedulerEndpoint>\"")
+    print("If you are using bash, run the following: export ENDPOINT=\"<schedulerEndpoint>\"")
+    exit()
+
+credential = DefaultAzureCredential()
+
+# Configure and start the worker
+with DurableTaskSchedulerWorker(host_address=endpoint, secure_channel=True,
+                                taskhub=taskhub_name, token_credential=credential) as w:
+    
+    w.add_orchestrator(orchestrator)
+    w.add_orchestrator(process_order)
+    w.add_activity(get_orders)
+    w.add_activity(check_and_update_inventory)
+    w.add_activity(charge_payment)
+    w.add_activity(ship_order)
+    w.add_activity(notify_customer)
+
+    w.start()
+
+    while True:
+        time.sleep(1)
```

Comment:
FYI for other reviewers:

This looks weird, but truly is the recommended way at this time:

https://github.com/grpc/grpc/blob/master/examples/python/route_guide/route_guide_server.py